src/storage/DataAccessorBasedStore.ts

```diff
@@ -468,4 +468,17 @@ export class DataAccessorBasedStore implements ResourceStore {
       }
     }
   }
+
+  public async resourceExists(identifier: ResourceIdentifier): Promise<boolean> {
+    try {
+      const result = await this.getRepresentation(identifier);
+      result.data.destroy();
+      return true;
+    } catch (error: unknown) {
+      if (NotFoundHttpError.isInstance(error)) {
+        return false;
+      }
+      throw error;
+    }
+  }
```

Comment:
The reason for adding it to this class is so we don't have to do this complex solution anymore. This entire block can be replaced with the beginning of the `getRepresentation` function:
```typescript
this.validateIdentifier(identifier);
await this.accessor.getMetadata(identifier);
```
This will throw a 404 if the resource doesn't exist.

Reply:
After writing I realize that of course we will still need to catch this 404 so a boolean can be returned.